### PR TITLE
PWA-2353 Use Product URL Suffix during PLP add to cart with configurables

### PR DESCRIPTION
**Pull request by JustinConabreeAbsolunet**
## Description
Using store config product URL suffix when redirecting to configurable product during add to cart in PLP

## Related Issue

Closes PWA-2353

#### Test scenario(s) for direct fix/feature

Given this backend: https://staging-vdt2zeq-c5v7sxvquxwl4.eu-4.magentosite.cloud/
- Test add to cart in PLP with a configurable product on the default store view
- Should be redirected to product page (**with** .html suffix)
- Test add to cart in PLP with a configurable product on the French store view
- Should be redirected to product page (**without** .html suffix)

## Checklist

- [x] I have added tests to cover my changes, if necessary.
- [ ] ~I have added translations for new strings, if necessary.~
- [ ] ~I have updated the documentation accordingly, if necessary.~

### CODE REVIEW

File: packages/venia-ui/lib/components/Gallery/addToCartButton.js [hunk 1]
```diff
@@ -24,7 +24,9 @@ const OutOfStockIcon = (
 );
 
 const AddToCartButton = props => {
+    const { urlSuffix } = props;
```
Comment (mikhaelbois) [inline packages/venia-ui/lib/components/Gallery/addToCartButton.js:27]: Small request.
Could we put `item` in this deconstruction too?

**Review by mikhaelbois — CHANGES_REQUESTED** [review]

File: packages/venia-ui/lib/components/Gallery/addToCartButton.js (hunk 1 above)
Comment (JustinConabreeAbsolunet) [reply packages/venia-ui/lib/components/Gallery/addToCartButton.js]: Fixed!

**Review by mikhaelbois — APPROVED** [review]
LGTM ✅

**Comment by fooman** [comment]
There was this recent PR https://github.com/magento/pwa-studio/pull/3393 which aimed to keep the url suffix configuration in one place. This PR seems to bring back querying for the configuration on separate graphQl calls.

**Comment by JustinConabreeAbsolunet** [comment]
> There was this recent PR #3393 which aimed to keep the url suffix configuration in one place. This PR seems to bring back querying for the configuration on separate graphQl calls.

Hi @fooman ,
This PR is passing that previous work down to the Add to Cart button on the PLP. In the PR you mentionned, there's [this file](https://github.com/magento/pwa-studio/pull/3393/files#diff-2cea6bf5bfea7076e495b19f4847bfb2988a817c12ef6cde90a2103635b9b45f) which sets the product link suffix. However the Add to Cart button was left untouched and hardcoded to a suffix of `.html`. Now we're just passing that suffix information from the item to the button component so it can set the correct suffix when it redirects. No additional GQL calls needed.

**Pull request merged**